src/main/java/com/google/api/codegen/transformer/SurfaceNamer.java

```diff
@@ -423,4 +423,20 @@ public String getGetResourceListCallName(Field resourcesField) {
   public String getAndSavePagedResponseTypeName(ModelTypeTable typeTable, TypeRef resourceType) {
     return getNotImplementedString("SurfaceNamer.getAndSavePagedResponseTypeName");
   }
+
+  public String getTestCaseName(Method method) {
+    return methodName(Name.upperCamel(method.getSimpleName(), "Test"));
+  }
+
+  public String getTestClassName(Interface service) {
+    return className(Name.upperCamel(service.getSimpleName(), "Test"));
+  }
+
+  public String getMockServiceClassName(Interface service) {
+    return className(Name.upperCamel("Mock", service.getSimpleName()));
+  }
+
+  public String getGetFunctionCallName(Name name) {
```

Comment:
Add comments on each of your new functions here (following the pattern of the rest of the file as most recently merged)